### PR TITLE
Distribute points questions: fix total points to distribute being reset to non-default value #8605

### LINKED ISSUE #8605: Distribute points questions: fix total points to distribute being reset to non-default value
**Environment**
`master` branch

**Steps to reproduce:**
- Login as Instuctor
- Create/Edit a session
- Add a `Distribute points` question. Does not matter Options/recipients
- Modify any of the `Total Points to distribute:` option, (The default is set as 100, delete it and click outside the box, or save the question)
- The value changes to 1

**Expected behaviour:** Don't know if it is expected or not, but IMO 1 point is not useful for any evaluation, specially when we are setting the value as 100 when adding a new question. 

**Actual behaviour:** Value changes to 1

**Suggested solutions:**
- Give a warning message when trying to save the changes before entering a value/ entering 0 (Like in MSQ question) : 
screenshot 405

- Set the default value as 100, so even if a user leaves the field blank / enters 0, it can be replaced to 100 (for consistency?)

**Additionally:** We should disable one field when the other is selected.
screenshot 406

@damithc Sir, your thoughts on this?

### PR DESCRIPTION
**Pull request by sukanta-27**
Fixes #8605 

**Outline of Solution**

 - [x] Set default value to 100
 - [x] Disable number field when radio button not selected

<s>**Status:** Ongoing</s>

### CODE REVIEW

**Comment by sukanta-27** [comment]
Ready for review.

File: src/main/webapp/dev/js/common/questionConstSum.js [hunk 1]
```diff
@@ -69,11 +69,34 @@ function removeConstSumOption(index, questionNum) {
...
+}
 
 export {
     addConstSumOption,
     hideConstSumOptionTable,
     removeConstSumOption,
     showConstSumOptionTable,
     updateConstSumPointsValue,
+    bindConstSumOptionsRadioButtons,
+    toggleConstSumOptionsRadioButton,
```
Comment (bqnguyen94) [inline src/main/webapp/dev/js/common/questionConstSum.js]: Can put these exports in alphabetical order?

File: src/main/webapp/dev/js/main/instructorFeedbackEdit.js [hunk 2]
```diff
@@ -45,6 +45,8 @@ import {
     removeConstSumOption,
     showConstSumOptionTable,
     updateConstSumPointsValue,
+    bindConstSumOptionsRadioButtons,
+    toggleConstSumOptionsRadioButton,
```
Comment (bqnguyen94) [inline src/main/webapp/dev/js/main/instructorFeedbackEdit.js]: Same here

File: src/test/java/teammates/test/cases/browsertests/FeedbackConstSumOptionQuestionUiTest.java [hunk 3]
```diff
@@ -230,4 +232,31 @@ private void testUiConsistencyForNewQuestion() {
         feedbackEditPage.waitForConfirmationModalAndClickOk();
 
     }
+
+    /**
+     * Test that the constSumPoints* number fields gets disabled, if the corresponding radio button is not checked.
+     */
+    private void testConstSumPointsDisableAction() {
```
Comment (bqnguyen94) [inline src/test/java/teammates/test/cases/browsertests/FeedbackConstSumOptionQuestionUiTest.java]: Nitpicking: `testDisableConstSumPoints` instead?

File: src/test/java/teammates/test/cases/browsertests/FeedbackConstSumOptionQuestionUiTest.java [hunk 4]
```diff
@@ -230,4 +232,31 @@ private void testUiConsistencyForNewQuestion() {
...
+        ______TS("Success case: CONSTSUM-option points to distribute field disables when radio button is unchecked");
+
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.selectNewQuestionType("CONSTSUM_OPTION");
+
+        //Make sure that constSumPointsTotal radio button is selected by default
+        assertTrue(browser.driver.findElement(By.id("constSumPointsTotal--1")).isSelected());
+        //verify that constSumPointsForEachOption field is disabled
```
Comment (bqnguyen94) [inline src/test/java/teammates/test/cases/browsertests/FeedbackConstSumOptionQuestionUiTest.java]: Nitpicking: Capitalise at start of comments? And space after `//`?

File: src/test/java/teammates/test/cases/browsertests/FeedbackConstSumRecipientQuestionUiTest.java [hunk 5]
```diff
@@ -148,4 +151,30 @@ public void testDeleteQuestionAction() {
...
+     */
+    private void testConstSumPointsDisableAction() {
+        ______TS("Success case: CONSTSUM-recipient points to distribute field disables when radio button is unchecked");
+
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.selectNewQuestionType("CONSTSUM_RECIPIENT");
+
+        //Make sure that constSumPointsTotal radio button is selected by default
+        assertTrue(browser.driver.findElement(By.id("constSumPointsTotal--1")).isSelected());
+        //verify that constSumPointsForEachRecipient field is disabled
```
Comment (bqnguyen94) [inline src/test/java/teammates/test/cases/browsertests/FeedbackConstSumRecipientQuestionUiTest.java]: Same nitpicking :P

File: src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java [hunk 6]
```diff
@@ -291,8 +291,8 @@ public void fillConstSumPointsBox(String points, int qnNumber) {
         String idSuffix = getIdSuffix(qnNumber);
 
         WebElement pointsBox = browser.driver.findElement(By.id("constSumPoints" + idSuffix));
-        // backspace to clear the extra 1 when box is cleared.
-        fillTextBox(pointsBox, Keys.RIGHT + " " + Keys.BACK_SPACE + points);
+        // backspace to clear the extra 100 when box is cleared.
```
Comment (bqnguyen94) [inline src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java]: Same here

File: src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java [hunk 7]
```diff
@@ -315,8 +315,8 @@ public void fillConstSumPointsForEachOptionBox(String points, int qnNumber) {
         String idSuffix = getIdSuffix(qnNumber);
 
         WebElement pointsBox = browser.driver.findElement(By.id("constSumPointsForEachOption" + idSuffix));
-        // backspace to clear the extra 1 when box is cleared.
-        fillTextBox(pointsBox, Keys.RIGHT + " " + Keys.BACK_SPACE + points);
+        // backspace to clear the extra 100 when box is cleared.
```
Comment (bqnguyen94) [inline src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java]: Same here

File: src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java [hunk 8]
```diff
@@ -339,8 +339,8 @@ public void fillConstSumPointsForEachRecipientBox(String points, int qnNumber) {
         String idSuffix = getIdSuffix(qnNumber);
 
         WebElement pointsBox = browser.driver.findElement(By.id("constSumPointsForEachRecipient" + idSuffix));
-        // backspace to clear the extra 1 when box is cleared.
-        fillTextBox(pointsBox, Keys.RIGHT + " " + Keys.BACK_SPACE + points);
+        // backspace to clear the extra 100 when box is cleared.
```
Comment (bqnguyen94) [inline src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java]: And same here

File: src/main/webapp/dev/js/main/instructorFeedbackEdit.js [hunk 9]
```diff
@@ -1217,6 +1223,7 @@ $(document).ready(() => {
 window.updateConstSumPointsValue = updateConstSumPointsValue;
 window.addConstSumOption = addConstSumOption;
 window.removeConstSumOption = removeConstSumOption;
+window.toggleConstSumOptionsRadioButton = toggleConstSumOptionsRadioButton;
```
Comment (bqnguyen94) [inline src/main/webapp/dev/js/main/instructorFeedbackEdit.js]: Is this used?

Comment (sukanta-27) [reply src/main/webapp/dev/js/main/instructorFeedbackEdit.js]: Oops. Sorry, it is not used. Removing it.

File: src/test/java/teammates/test/cases/browsertests/FeedbackConstSumOptionQuestionUiTest.java (hunk 3 above)
Comment (sukanta-27) [reply src/test/java/teammates/test/cases/browsertests/FeedbackConstSumOptionQuestionUiTest.java]: How about `testDisableConstSumPointsAction()` instead? To make it similar to `testAddQuestionAction()` etc tests?

**Comment by sukanta-27** [comment]
@bqnguyen94 Ready for review.

**Comment by bqnguyen94** [comment]
Good job @sukanta-27 👍
@whipermr5 LGTM. Should I assign a @TEAMMATES/seniordevs ?

File: src/test/java/teammates/test/cases/browsertests/FeedbackConstSumRecipientQuestionUiTest.java [hunk 10]
```diff
@@ -148,4 +151,30 @@ public void testDeleteQuestionAction() {
         assertNull(BackDoor.getFeedbackQuestion(courseId, feedbackSessionName, 1));
     }
 
+    /**
+     * Test that the constSumPoints* number fields gets disabled, if the corresponding radio button is not checked.
```
Comment (damithc) [inline src/test/java/teammates/test/cases/browsertests/FeedbackConstSumRecipientQuestionUiTest.java]: `Tests`

File: src/main/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetails.java [hunk 11]
```diff
@@ -92,9 +89,15 @@ public boolean extractQuestionDetails(
...
         if (pointsPerOption) {
-            points = distributeToRecipients ? Integer.parseInt(pointsForEachRecipientString)
-                                            : Integer.parseInt(pointsForEachOptionString);
+            if (distributeToRecipients) {
+                Assumption.assertNotNull("Null points for each recipient", pointsForEachRecipientString);
+                points = Integer.parseInt(pointsForEachRecipientString);
+            } else {
+                Assumption.assertNotNull("Null points for each option", pointsForEachOptionString);
+                points = Integer.parseInt(pointsForEachOptionString);
+            }
```
Comment (damithc) [inline src/main/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetails.java]: Something like this is more concise?
```java
String someSuitalbeName = distributeToRecipients? pointsForEachRecipientString : pointsForEachOptionString;
Assumption.assertNotNull("Null points for each recipient", someSuitalbeName);
points = Integer.parseInt(someSuitalbeName);
```

Comment (sukanta-27) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetails.java]: @damithc Sir, how about the name `pointsForEachOption` for the variable? As `pointsForEachRecipientString` and `pointsForEachOptionString` are both at its core, points for each option only?

Comment (damithc) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetails.java]: We need a name that represents both `pointsForEachRecipientString` and `pointsForEachOptionString` without implying either one.
just `pointsString`?

Comment (sukanta-27) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetails.java]: We are already using `pointsString` for `Total points to distribute` field.

Comment (sukanta-27) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetails.java]: @damithc I guess we can rename the variable name for `Total points to distribute` from `pointsString` to `pointsTotalString`, and use `pointsString` for the above case?

Comment (damithc) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackConstantSumQuestionDetails.java]: Sure. although `totalPointsString` sounds easier read.

**Comment by sukanta-27** [comment]
@damithc Sir, ready for review.

**Comment by sukanta-27** [comment]
@whipermr5 A reminder to merge. :sweat_smile:

**Pull request merged**